content/blog/getting-started-with-deno/puppeteer-with-deno.md

```diff
@@ -0,0 +1,217 @@
+});
+
+const page = await browser.newPage();
+
+// website is a cli parameter
+await page.goto(website, {
+  waitUntil: "networkidle2",
+});
+
+const screenshot = await page.screenshot();
+```
+
+After having a screenshot, it is just a matter or saving it on the filesystem or comparing it with the one previously stored.
+
+To save to the file system we'd just use `Deno.writeFile`. To compare with the previously saved image we'll use `pixelmatch` and `pngjs`, as shown by the script below.
+
+
+```ts
+import pixelmatch from "https://jspm.dev/pixelmatch";
+import { PNG } from "https://jspm.dev/pngjs";
+
+// Cut for brevity
+
+const newImage = await parsePNG(screenshot as Uint8Array);
+const oldImage = await parsePNG(await Deno.readFile(screenshotPath));
+const diff = new PNG(viewPort);
+
+pixelmatch(
+  newImage.data,
+  oldImage.data,
+  diff.data,
+  viewPort.width,
+  viewPort.height,
+  { threshold: 0.5 },
+);
+
+const buffer = PNG.sync.write(diff);
+await Deno.writeFile(
+  // domain is the website domain, resolution the screen resolution
+  `${screenshotBasePath}/diff-${domain}-${resolution}.png`,
+  buffer,
+  { create: true },
+);
+```
+
+The full code is available [here](https://github.com/asantos00/deno-website-diff).
+
+Again, the code is quite straightforward. We're using Deno runtime APIs to access the filesystem, and the third-party packages to decode and compare both images.
+
+If the Deno APIs look strange to you, fear nothing, Deno's documentation is quite complete on that.
+
+## Documentation
+
+Any questions regarding the available APIs and how they work is delighfully explained in Deno's documentation (https://doc.deno.land/).
+
+Deno supports directly importing modules by URL in the code, and the documentation isn't any different. By using the provided URL (https://doc.deno.land) plus the URL to a package, we get a beautifully designed page with the package types and documentation.
+
+For instance, to check `readFile` documentation we just need to navigate to https://doc.deno.land/builtin/stable#Deno.readFile and this is what we get.
+
+![readfile-documentation](./deno-read-file.png)
+
+*This is possible to be used with a direct GitHub link to any TS file, ex: https://doc.deno.land/https/raw.githubusercontent.com/timonson/djwt/master/mod.ts*
+
+## Executing the code
+
+After having Deno installed on the system, executing the code is the simplest of tasks. We can just use `deno run` plus the path to the file we want to execute (in our case we've called it `mod.ts`).
+
+However, and as we previously mentioned, all Deno programs run in a sandbox thus we need to give your program the specific permissions it needs to execute.
+
+For our specific usecase, and because puppeteer needs quite a lot of permissions, it needs to access the environment, network, file system, and have the ability to run processes.
+
+We'll also have to use the `--unstable` flag to enable our program to access a few unstable Deno APIs.
```

Comment:
Maybe interesting to just briefly explain which unstable APIs and the fact that it is due to notify users that these features are still under active development. Just so that people do not get thrown off by the fact that deno is sold as:
- Core security features (but then skip/allow all stuff)
- Nice and well written language (but then flag it as unstable)